Feat/image: Image 컴포넌트 구현

iyu88 opened the pull request:
size 와 shape 을 결정할 수 있는 Image 컴포넌트를 구현했습니다.

## 🤠 개요

- #12 
- 프로필 이미지나 포인트 이미지 등으로 활용할 수 있는 이미지 컴포넌트를 구현했습니다. 

## 💫 설명

- Image Props 설명 

  - src : 이미지 URL 이나 경로

  - alt : 이미지 설명

  - size : small, medium, large 중 택 1

  - shape : circle, rounded 중 택 1

- 로컬에서 스토리북 실행이 안되서 확인을 못했어요 ( 대신 확인해주시면 감사합니다 )

- 기존에는 React-icon 도 표시할 수 있는 컴포넌트로 만들려고 했지만, 논의 끝에 해당 기능은 Button 컴포넌트에서 텍스트와 함께 구현하는 것으로 결정되었습니다. 

## 📷 스크린샷 (Optional)
![image](https://user-images.githubusercontent.com/46566891/220117278-afb3a33e-0f07-4ab3-b491-0c7aa09c9c42.png)
![image](https://user-images.githubusercontent.com/46566891/220117295-e2c38606-cce5-415d-bdd2-71d9565a73ae.png)
![image](https://user-images.githubusercontent.com/46566891/220117310-83ba92e6-415c-4627-8a22-c38890882548.png)

iyu88 committed:
feat: #12 Image 컴포넌트 구현

size 와 shape 을 결정할 수 있는 Image 컴포넌트를 구현했습니다.

se030 commented on src/components/Image/Image.stories.tsx:
```diff
@@ -0,0 +1,85 @@
+const DUMMY_SRC = 'https://picsum.photos/200/300';
+const DUMMY_ALT = '테스트 이미지를 설명';
+
+const Template: ComponentStory<typeof Image> = (args) => <Image {...args} />;
```
```ts
<Image src={DUMMY_SRC} alt={DUMMY_ALT}  {...args} />
```
하면 아래에서 동일한 부분은 반복적으로 작성하지 않아도 돼요! 각각의 스토리가 의미하는바가 더 잘 드러날 것 같아요.

iyu88 replied:
안그래도 불편했는데 하나 배웠습니다 👍

se030 commented on src/components/Image/index.tsx:
```diff
@@ -0,0 +1,29 @@
+type ImageShapeProps = `${ImageShapeVariant}`;
+
+interface ImageProps {
+  src: string;
```
그때 기존 속성이랑 이름이 겹쳐서 그런지 `fill` 같은게 콘솔에서 에러 났었는데, 얘는 괜찮을지 모르겠지만 DefaultProps 확장해서 사용하면 더 좋지 않을까요?

iyu88 replied:
반영했습니다 👍

leesunmin1231 commented:
storybook 실행 해봤는데 잘 동작해요! 굿굿

dohun31 commented:
스크린샷이 필요해요

se030 committed:
Feat/util-types: DefaultProps 유틸리티 타입 추가 (#14)

* feat: DefaultProps<T> 유틸리티 타입 추가

HTMLElement T 리액트 노드가 가질 수 있는 속성들과 emotion의 css를 포함

* feat: DefaultPropsWithChildren<T> 유틸리티 타입 추가

- children을 꼭 가져야하는 컴포넌트에서 사용
- https://velog.io/@kkojae91/PropsWithChildren%EB%8A%94-%EC%95%88%EC%A0%84%ED%95%9C-%ED%83%80%EC%9E%85%EC%9D%BC%EA%B9%8C

* feat: utils, util-types 절대경로 설정 추가

- 유틸리티 타입을 위해 `src/utils`와 하위 types 디렉토리 생성
- `@types` 디렉토리 아래에서 .d.ts 파일에 선언해봤지만 적용이 되지 않음
- `@types/utils` 만들면 절대경로 설정 시 "Cannot import type declaration files. ts(6137)" 오류 발생
- react에서도 PropsWithChildren 등 유틸리티 타입들을 named export로 제공하고 있어 동일한 방식 사용

se030 committed:
Fix: DefaultProps에 optional children 추가 (#20)

iyu88 committed:
fix: Image 스토리에 공통된 속성 삭제

iyu88 committed:
fix: ImageProps 에서 DefaultProps 사용
